docs/learn/multi-format-reader.md

```diff
@@ -0,0 +1,263 @@
+
+Here, we will explain the inner workings of the `MultiFormatReader`. Note that there is also a [how-to guide](../how-tos/use-multi-format-reader.md) on how to implement a new reader off of the `MultiFormatReader` using a concrete example. In case you simply want to use the `MultiFormatReader` without understanding its inner logic, we recommend you start there.
+
+## The basic structure
+
+For extending the `MultiFormatReader`, the following basic structure must be implemented:
+```python
+"""MyDataReader implementation for the DataConverter to convert mydata to NeXus."""
+from typing import Tuple, Any
+
+from pynxtools.dataconverter.readers.base.reader import MultiFormatReader
+
+class MyDataReader(MultiFormatReader):
+    """MyDataReader implementation for the DataConverter to convert mydata to NeXus."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.extensions = {
+            ".yml": self.handle_eln_file,
+            ".yaml": self.handle_eln_file,
+            ".json": self.set_config_file,
+            # Here, one must add functions for handling any other file extension(s)
+        }
+# This has to be set to allow the convert script to use this reader. Set it to "MyDataReader".
+READER = MyDataReader
+```
+
+In order to understand the capabilities of the `MultiFormatReader` and which methods need to be implemented when extending it, we will have a look at its ```read``` method:
+```python
+def read(
+    self,
+    template: dict = None,
+    file_paths: Tuple[str] = None,
+    objects: Optional[Tuple[Any]] = None,
+    **kwargs,
+) -> dict:
+    self.kwargs = kwargs
+    self.config_file = self.kwargs.get("config_file", self.config_file)
+    self.overwrite_keys = self.kwargs.get("overwrite_keys", self.overwrite_keys)   
+```
+### Template initialization and processing order
+An empty `Template` object is initialized that later gets filled from the data files later.
+```python
+    template = Template(overwrite_keys=self.overwrite_keys)
+
+    def get_processing_order(path: str) -> Tuple[int, Union[str, int]]:
+        """
+        Returns the processing order of the file.
+        """
+        ext = os.path.splitext(path)[1]
+        if self.processing_order is None or ext not in self.processing_order:
+            return (1, ext)
+        return (0, self.processing_order.index(ext))
+
+    sorted_paths = sorted(file_paths, key=get_processing_order)
+```
+If the reader has a `self.processing_order`, the input files get sorted in this order. If `self.overwrite_keys` is True, later files get precedent.

```

Comment:
Maybe give an example for a processing_order entry.

Reply:
added